### ISSUE TITLE
linux-pam missing build dependency rpc/rpc.h

### ISSUE THREAD

**Issue by vsoch** [LLNL/spack#40419]
We are trying to build with the new ubuntu:jammy and running into this issue:

![image](https://github.com/spack/spack/assets/814322/765fac9f-0f97-4285-88bd-3c18f8b89e30)

It happens with linux-pam and you can see the full log here: https://github.com/converged-computing/flux-views/actions/runs/6464329605/job/17548802780

Also posted here for provenance when that goes away:

```console
#10 2373.6   Stage: 0.87s.  Autoreconf: 5.99s.  Configure: 15.63s.  Build: 44.05s.  Install: 0.61s.  Post-install: 0.06s.  Total: 1m 7.32s
#10 2373.6 [+] /opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/gmp-6.2.1-fgu56ebmxlcvwqldyv4tv6beb4yh2jbd
#10 2373.7 ==> Installing linux-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj [60/99]
#10 2373.7 ==> No binary for linux-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj found: installing from source
#10 2374.2 ==> Fetching https://mirror.spack.io/_source-cache/archive/e4/e4ec7131a91da44512574268f493c6d8ca105c87091691b8e9b56ca685d4f94d.tar.xz
#10 2374.2 ==> No patches needed for linux-pam
#10 2374.3 ==> linux-pam: Executing phase: 'autoreconf'
#10 2374.3 ==> linux-pam: Executing phase: 'configure'
#10 2385.1 ==> linux-pam: Executing phase: 'build'
#10 2401.7 ==> Error: ProcessError: Command exited with status 2:
#10 2401.7     'make' '-j2' 'V=1'
#10 2402.0 
#10 2402.0 5 errors found in build log:
#10 2402.0      925    libtool: compile:  /opt/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H
#10 2402.0              -I. -I../.. -I../../libpam/include -I../../libpamc/include -DCHKPW
#10 2402.0             D_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linux-
#10 2402.0             pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_chkpwd\" -DUPD
#10 2402.0             ATE_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linu
#10 2402.0             x-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_update\" -W 
#10 2402.0             -Wall -Wbad-function-cast -Wcast-align -Wcast-align=strict -Wcast-q
#10 2402.0             ual -Wdeprecated -Winline -Wmain -Wmissing-declarations -Wmissing-f
#10 2402.0             ormat-attribute -Wmissing-prototypes -Wpointer-arith -Wreturn-type 
#10 2402.0             -Wshadow -Wstrict-prototypes -Wuninitialized -Wwrite-strings -g -O2
#10 2402.0              -MT pam_unix_auth.lo -MD -MP -MF .deps/pam_unix_auth.Tpo -c pam_un
#10 2402.0             ix_auth.c  -fPIC -DPIC -o .libs/pam_unix_auth.o
#10 2402.0      926    mv -f .deps/pam_unix_acct.Tpo .deps/pam_unix_acct.Plo
#10 2402.0      927    /bin/bash ../../libtool  --tag=CC   --mode=compile /opt/spack/lib/s
#10 2402.0             pack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I../..    -I../../libpam/incl
#10 2402.0             ude -I../../libpamc/include -DCHKPWD_HELPER=\"/opt/software/linux-u
#10 2402.0             buntu22.04-x86_64/gcc-11.4.0/linux-pam-1.5.2-7hao7wthb4iysy6lliuiim
#10 2402.0             fbdg6tmpkj/sbin/unix_chkpwd\" -DUPDATE_HELPER=\"/opt/software/linux
#10 2402.0             -ubuntu22.04-x86_64/gcc-11.4.0/linux-pam-1.5.2-7hao7wthb4iysy6lliui
#10 2402.0             imfbdg6tmpkj/sbin/unix_update\" -W -Wall -Wbad-function-cast -Wcast
#10 2402.0             -align -Wcast-align=strict -Wcast-qual -Wdeprecated -Winline -Wmain
#10 2402.0              -Wmissing-declarations -Wmissing-format-attribute -Wmissing-protot
#10 2402.0             ypes -Wpointer-arith -Wreturn-type -Wshadow -Wstrict-prototypes -Wu
#10 2402.0             ninitialized -Wwrite-strings -g -O2 -MT pam_unix_passwd.lo -MD -MP 
#10 2402.0             -MF .deps/pam_unix_passwd.Tpo -c -o pam_unix_passwd.lo pam_unix_pas
#10 2402.0             swd.c
#10 2402.0      928    libtool: compile:  /opt/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H
#10 2402.0              -I. -I../.. -I../../libpam/include -I../../libpamc/include -DCHKPW
#10 2402.0             D_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linux-
#10 2402.0             pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_chkpwd\" -DUPD
#10 2402.0             ATE_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linu
#10 2402.0             x-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_update\" -W 
#10 2402.0             -Wall -Wbad-function-cast -Wcast-align -Wcast-align=strict -Wcast-q
#10 2402.0             ual -Wdeprecated -Winline -Wmain -Wmissing-declarations -Wmissing-f
#10 2402.0             ormat-attribute -Wmissing-prototypes -Wpointer-arith -Wreturn-type 
#10 2402.0             -Wshadow -Wstrict-prototypes -Wuninitialized -Wwrite-strings -g -O2
#10 2402.0              -MT pam_unix_passwd.lo -MD -MP -MF .deps/pam_unix_passwd.Tpo -c pa
#10 2402.0             m_unix_passwd.c  -fPIC -DPIC -o .libs/pam_unix_passwd.o
#10 2402.0      929    mv -f .deps/pam_unix_auth.Tpo .deps/pam_unix_auth.Plo
#10 2402.0      930    /bin/bash ../../libtool  --tag=CC   --mode=compile /opt/spack/lib/s
#10 2402.0             pack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I../..    -I../../libpam/incl
#10 2402.0             ude -I../../libpamc/include -DCHKPWD_HELPER=\"/opt/software/linux-u
#10 2402.0             buntu22.04-x86_64/gcc-11.4.0/linux-pam-1.5.2-7hao7wthb4iysy6lliuiim
#10 2402.0             fbdg6tmpkj/sbin/unix_chkpwd\" -DUPDATE_HELPER=\"/opt/software/linux
#10 2402.0             -ubuntu22.04-x86_64/gcc-11.4.0/linux-pam-1.5.2-7hao7wthb4iysy6lliui
#10 2402.0             imfbdg6tmpkj/sbin/unix_update\" -W -Wall -Wbad-function-cast -Wcast
#10 2402.0             -align -Wcast-align=strict -Wcast-qual -Wdeprecated -Winline -Wmain
#10 2402.0              -Wmissing-declarations -Wmissing-format-attribute -Wmissing-protot
#10 2402.0             ypes -Wpointer-arith -Wreturn-type -Wshadow -Wstrict-prototypes -Wu
#10 2402.0             ninitialized -Wwrite-strings -g -O2 -MT pam_unix_sess.lo -MD -MP -M
#10 2402.0             F .deps/pam_unix_sess.Tpo -c -o pam_unix_sess.lo pam_unix_sess.c
#10 2402.0   >> 931    pam_unix_passwd.c:80:11: fatal error: rpc/rpc.h: No such file or di
#10 2402.0             rectory
#10 2402.0      932       80 | # include <rpc/rpc.h>
#10 2402.0      933          |           ^~~~~~~~~~~
#10 2402.0      934    compilation terminated.
#10 2402.0      935    libtool: compile:  /opt/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H
#10 2402.0              -I. -I../.. -I../../libpam/include -I../../libpamc/include -DCHKPW
#10 2402.0             D_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linux-
#10 2402.0             pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_chkpwd\" -DUPD
#10 2402.0             ATE_HELPER=\"/opt/software/linux-ubuntu22.04-x86_64/gcc-11.4.0/linu
#10 2402.0             x-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/sbin/unix_update\" -W 
#10 2402.0             -Wall -Wbad-function-cast -Wcast-align -Wcast-align=strict -Wcast-q
#10 2402.0             ual -Wdeprecated -Winline -Wmain -Wmissing-declarations -Wmissing-f
#10 2402.0             ormat-attribute -Wmissing-prototypes -Wpointer-arith -Wreturn-type 
#10 2402.0             -Wshadow -Wstrict-prototypes -Wuninitialized -Wwrite-strings -g -O2
#10 2402.0              -MT pam_unix_sess.lo -MD -MP -MF .deps/pam_unix_sess.Tpo -c pam_un
#10 2402.0             ix_sess.c  -fPIC -DPIC -o .libs/pam_unix_sess.o
#10 2402.0   >> 936    make[3]: *** [Makefile:879: pam_unix_passwd.lo] Error 1
#10 2402.0      937    make[3]: *** Waiting for unfinished jobs....
#10 2402.0      938    mv -f .deps/pam_unix_sess.Tpo .deps/pam_unix_sess.Plo
#10 2402.0      939    make[3]: Leaving directory '/tmp/root/spack-stage/spack-stage-linux
#10 2402.0             -pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/spack-src/modules/pam_u
#10 2402.0             nix'
#10 2402.0   >> 940    make[2]: *** [Makefile:495: all-recursive] Error 1
#10 2402.0      941    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-linux
#10 2402.0             -pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/spack-src/modules'
#10 2402.0   >> 942    make[1]: *** [Makefile:496: all-recursive] Error 1
#10 2402.0      943    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-linux
#10 2402.0             -pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/spack-src'
#10 2402.0   >> 944    make: *** [Makefile:426: all] Error 2
#10 2402.0 
#10 2402.0 See build log for details:
#10 2402.0   /tmp/root/spack-stage/spack-stage-linux-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj/spack-build-out.txt
#10 2402.0 
#10 2402.0 ==> Warning: Skipping build of libpam-1.09-vkrf6x6qcrxliit27afjfiuezrl4gc4l since linux-pam-1.5.2-7hao7wthb4iysy6lliuiimfbdg6tmpkj failed
#10 2402.0 ==> Warning: Skipping build of flux-security-0.10.0-3rubbxdztabgjf2l6cggef4afzkwpfa4 since libpam-1.09-vkrf6x6qcrxliit27afjfiuezrl4gc4l failed
#10 2404.8 ==> Error: Terminating after first install failure: ProcessError: Command exited with status 2:
#10 2404.8     'make' '-j2' 'V=1'
#10 2404.8 ==> Updating view at /opt/views/view
#10 ERROR: process "docker-shell cd /opt/spack-environment && spack env activate . && spack install --fail-fast && spack gc -y" did not complete successfully: exit code: 1
```

This is for the version of spack on develop yesterday evening, and the ubuntu jammy base image at the same time.

**Comment by bernhardkaindl** [comment]
@vsoch This should be fixed by #40459! (Needs a new version of PAM as `rpc/rpc.h` is dead)

**Comment by vsoch** [comment]
>  (Needs a new version of PAM as rpc/rpc.h is dead)

ohno :skull: 

I'll watch that, thank you @bernhardkaindl!

**Comment by bernhardkaindl** [comment]
Should be fixed by #45344
